rxdiffadapter/src/main/java/com/revolut/rxdiffadapter/RxDiffAdapter.kt

```diff
@@ -141,9 +143,18 @@ open class RxDiffAdapter @Deprecated("Replace with constructor without delegates
             diffResult.dispatchUpdatesTo(this)
         }
 
+        val lastCallback = lastDispatchDiffCallback
+        if (lastCallback != null) {
+            rv.removeCallbacks(lastCallback)
+        }
         if (rv.isComputingLayout) {
-            rv.post { dispatchDiff() }
+            val newDispatchDiffCallback = Runnable {
+                dispatchDiff()
+            }
+            lastDispatchDiffCallback = newDispatchDiffCallback
+            rv.post(newDispatchDiffCallback)
         } else {
+            lastDispatchDiffCallback = null
```

Comment:
let's move this statement into dispatchDiff() method. Then it would be invoked in both cases